### PR TITLE
LPS-63213 Fix names/ids collisions because we are supposing preffix--name-- as pattern in names and sometimes this is not true

### PR DESCRIPTION
**Pull request by JorgeFerrer**
/cc @jbalsas

### CODE REVIEW

**Comment by jbalsas** [comment]
👍

**Comment by jbalsas** [comment]
I understand where this is coming from, but the fix is hard to follow once applied... since this applies an additional branch, maybe we could add an extra param to the taglib (`preserveName`) which defaults to `false`, so everything works the same, but that we could set to `true` in the instances we need so it's clear that we don't want the "magic" behaviour.

**Comment by jbalsas** [comment]
On the other hand, the fix could also happen on the application level, where you won't use the `--` prefix in the name as this ticket is currently suggesting.

**Comment by JorgeFerrer** [comment]
Closing to discuss it further with @jbalsas
